scripts/ts/checkOutdatedDependencies/checkOutdatedDependencies.ts

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-console */
+import { exec } from "child_process";
+import * as util from "util";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import * as semver from "semver";
+import { slackPostMessage } from "../common/slack/postMessage";
+import { generateSlackMessage } from "./generateSlackMessage";
+import { initOutdatedStats } from "./types/defaultValues";
+import { DependenciesTable } from "./types/DependeciesTable";
+import { getSeverityType } from "./types/GroupBySeverity";
+import { getDependencyType, increaseSeverityAmount } from "./types/GroupByType";
+import { OutdatedStats, toOutdatedPackage } from "./types/OutdatedStats";
+
+const execAsync = util.promisify(exec);
+
+const majorVersionDeltaWarning = 3;
+
+/**
+ * Execute the command "yarn outdated --json" and return the results as JSON any
+ */
+const readOutdatedJson = async (): Promise<any> => {
+  try {
+    const std = await execAsync("yarn outdated --json");
+    return JSON.parse(std.stdout.split("\n")[1]);
+  } catch (error) {
+    // When the command finds outdated packages, return with exit code 1
+    if (error.code === 1) {
+      return JSON.parse(error.stdout.split("\n")[1]);
```

Comment:
Maybe we are always sure that this array has a length >=2
But it can produce a `SyntaxError: Unexpected token u in JSON at position 0` if the array has a length < 2. `JSON.parse` will be invoked with `undefined` parameter

What do you think about this safe access?
```suggestion
    const splittedErrorStdout = error.stdout.split("\n");
    // When the command finds outdated packages, return with exit code 1
    if (error.code === 1 && splittedErrorStdout.length > 1) {
      return JSON.parse(splittedErrorStdout[1]);
```

Reply:
Atm the size is always the same because the first block contains the header of the table and the second the table. But of course can change in the future, better be safe than sorry! Changed in b4ae006efb6d2b68ed3d5142e507165d4ea8ff6a with a little refactoring to avoid duplicate code (the same check should be done when no error code are returned.